src/rules/unit-blacklist/index.js

```diff
@@ -0,0 +1,43 @@
+import { isString } from "lodash"
+import valueParser from "postcss-value-parser"
+
+import {
+  declarationValueIndexOffset,
+  report,
+  ruleMessages,
+  validateOptions,
+} from "../../utils"
+
+export const ruleName = "unit-blacklist"
+
+export const messages = ruleMessages(ruleName, {
+  rejected: (p) => `Unexpected unit "${p}"`,
```

Comment:
Minor nitpick, but I think `p` stood for `property` in `property-blacklist`, so you might want to change this to `u` or `unit`.